sdk/confidentialledger/Azure.Security.CodeTransparency/tests/SamplesUnitTests.cs

```diff
@@ -0,0 +1,306 @@
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.TestFramework;
+using Azure.Identity;
+using Azure.Security.CodeTransparency.Receipt;
+using System.Security.Cryptography.Cose;
+using NUnit.Framework;
+
+namespace Azure.Security.CodeTransparency.Tests
+{
+    public class SamplesUnitTests: ClientTestBase
+    {
+        private string _fileQualifierPrefix;
+
+        private byte[] readFileBytes(string name) {
+            var assembly = Assembly.GetExecutingAssembly();
+            using (Stream stream = assembly.GetManifestResourceStream(_fileQualifierPrefix+name))
+            using (MemoryStream mem = new())
+            {
+                if (stream == null)
+                    throw new FileNotFoundException("Resource not found: " + _fileQualifierPrefix+name);
+                stream.CopyTo(mem);
+                return mem.ToArray();
+            }
+        }
+
+        public SamplesUnitTests(bool isAsync) : base(isAsync)
+        {
+        }
+
+        [SetUp]
+        public void BaseSetUp() {
+            var assembly = Assembly.GetExecutingAssembly();
+            string mustExistFilename = "sbom.json";
+            string resourceName = assembly.GetManifestResourceNames().Single(str => str.EndsWith(mustExistFilename));
+            Assert.IsNotNull(resourceName);
+            _fileQualifierPrefix = resourceName.Split(new String[]{mustExistFilename}, StringSplitOptions.None)[0];
+        }
+
+        [Test]
+        public async Task Snippet_Readme_CodeTransparencySubmission_Test()
+        {
+            var createResponse = new MockResponse(200);
+            createResponse.SetContent("{\"operationId\": \"foobar\"}");
+            var pendingResponse = new MockResponse(200);
+            pendingResponse.SetContent("{\"operationId\": \"foobar\", \"status\": \"running\"}");
+            var operationResponse = new MockResponse(200);
+            operationResponse.SetContent("{\"operationId\": \"foobar\", \"entryId\": \"123.23\", \"status\": \"succeeded\"}");
+            var entryResponse = new MockResponse(200);
+            entryResponse.AddHeader("Content-Type", "application/cose");
+            entryResponse.SetContent(new byte[] { 0x01, 0x02, 0x03 });
+            var mockTransport = new MockTransport(createResponse, pendingResponse, operationResponse, entryResponse);
+            var options = new CodeTransparencyClientOptions
+            {
+                Transport = mockTransport,
+                IdentityClientEndpoint = "https://foo.bar.com"
+            };
+
+            #region Snippet:CodeTransparencySubmission
+#if !SNIPPET
+            CodeTransparencyClient client = new(new Uri("https://foo.bar.com"), new AzureKeyCredential("token"), options);
+#endif
+#if SNIPPET
+            CodeTransparencyClient client = new(new Uri("https://cts-service.confidential-ledger.azure.com"), null);
```

Comment:
We can't commit this as-is; it'll get flagged by scanning as a leak of sensitive information.   If it's an example, we'll need to make it look more obscure.   

If we need this for test runs, then we'll want to read from an environment variable.   But, this should be marked as a Live test if it requires access to Azure resources.

Reply:
The ones for the snippet replaced to have a slug, `foo.bar.com` left intact. Please confirm if this was the expectation. FYI the generated tests use `my-service.azure.com`